src/react-native-animated-charts/src/charts/linear/ChartPath.tsx

```diff
@@ -0,0 +1,375 @@
+import React, { useEffect } from 'react';
+import { Platform, View, ViewProps } from 'react-native';
+import {
+  LongPressGestureHandler,
+  LongPressGestureHandlerGestureEvent,
+  LongPressGestureHandlerProperties,
+} from 'react-native-gesture-handler';
+import ReactNativeHapticFeedback from 'react-native-haptic-feedback';
+import Animated, {
+  cancelAnimation,
+  runOnJS,
+  runOnUI,
+  useAnimatedGestureHandler,
+  useAnimatedProps,
+  useAnimatedReaction,
+  useAnimatedStyle,
+  useSharedValue,
+  useWorkletCallback,
+  withDelay,
+  withTiming,
+} from 'react-native-reanimated';
+import { getYForX } from 'react-native-redash';
+import Svg, { Path, PathProps } from 'react-native-svg';
+import { PathData } from '../../helpers/ChartContext';
+import {
+  requireOnWorklet,
+  useWorkletValue,
+} from '../../helpers/requireOnWorklet';
+import { useChartData } from '../../helpers/useChartData';
+
+export const FIX_CLIPPED_PATH_MAGIC_NUMBER = 22;
+
+function ascending(a?: number, b?: number) {
+  'worklet';
+
+  return a == null || b == null
+    ? NaN
+    : a < b
+    ? -1
+    : a > b
+    ? 1
+    : a >= b
+    ? 0
+    : NaN;
+}
+
+function least(length: number, compare: typeof ascending = ascending) {
+  'worklet';
+
+  let min;
+  let defined = false;
+
+  let minValue;
+  for (let i = 0; i < length; i++) {
+    const value = compare(i);
+    if (
+      defined ? ascending(value, minValue) < 0 : ascending(value, value) === 0
+    ) {
+      min = i;
+      minValue = value;
+      defined = true;
+    }
+  }
+
+  return min;
+}
```

Comment:
As far as I understand, we use this function only in a single place for finding a point that is closest to some value. 

`compare` function returns a distance from `i` point to the one point over the x-axis.

![image](https://user-images.githubusercontent.com/25709300/146210905-48505634-3a5b-4a97-a526-9e4e6f5540e5.png)

So I expect the results will model this function with precision to qualification and a displacement over the x-axis.
We are willing to find a point that is the closes one to the "dip" (the root of this function when this touches the x-axis).

Why can't we use a divide-and-conquer bisection search for this? Complexity of this is `O(n)` where `n = length`.
Bisection can live us `O(log n)` that is nearly constant. 




```suggestion
function least(length: number, compare: typeof ascending = ascending) {
  'worklet';
  
  let bound1 = 0;
  let = length - 1;
  
  
  while(true) {
  	 let pivot = Math.round(bound1 + (bound2 - bound1) / 2)
  	 if (pivot === bound1) {
  	 	return bound1;
  	 } 
	  if (pivot === bound2) {
  	 	return bound2;
  	 }

  	 if (compare(pivot - 1) - compare(pivor) > 0) {
  	 	// decreasing, dip on the right side
  	 	bound1 = pivot;
  	 } else {
  	 	// non-increasing or dip, dip on the left side or in pivot
  	 	bound2 = pivot
  	 }
     }	
  }

```